src/DataTypes/Serializations/SerializationNumber.cpp

```diff
@@ -21,12 +21,17 @@ void SerializationNumber<T>::serializeText(const IColumn & column, size_t row_nu
 }
 
 template <typename T>
-void SerializationNumber<T>::deserializeText(IColumn & column, ReadBuffer & istr, const FormatSettings &) const
+void SerializationNumber<T>::deserializeText(IColumn & column, ReadBuffer & istr, const FormatSettings & settings) const
 {
     T x;
 
     if constexpr (is_integer_v<T> && is_arithmetic_v<T>)
-        readIntTextUnsafe(x, istr);
+    {
+        if (settings.read_int_safe)
```

Comment:
This branch alone can introduce some cost.

Reply:
But it's already in virtual method. I think it's not such significant.